### PR TITLE
hepmc3: add v3.3.0

### PR DESCRIPTION
**Pull request by wdconinc**
This PR adds HepMC3 v3.3.0, which (as only relevant change) deduces the C++ standard that ROOT was built with, and uses it for the HepMC3 build. Since that is now provided by the package, we don't have to force the standard.

Test build:
```
==> Installing hepmc3-3.3.0-gocuipxl3qyy4imx3fafl4pbsaho7veo [75/75]
==> No binary for hepmc3-3.3.0-gocuipxl3qyy4imx3fafl4pbsaho7veo found: installing from source
==> Fetching https://gitlab.cern.ch/hepmc/HepMC3/-/archive/3.3.0/HepMC3-3.3.0.tar.gz
==> No patches needed for hepmc3
==> hepmc3: Executing phase: 'cmake'
==> hepmc3: Executing phase: 'build'
==> hepmc3: Executing phase: 'install'
==> hepmc3: Successfully installed hepmc3-3.3.0-gocuipxl3qyy4imx3fafl4pbsaho7veo
  Stage: 6.00s.  Cmake: 1.13s.  Build: 1m 4.62s.  Install: 0.12s.  Post-install: 0.33s.  Total: 1m 12.51s
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/hepmc3-3.3.0-gocuipxl3qyy4imx3fafl4pbsaho7veo
```

### CODE REVIEW

**Review by jmcarcell — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256.

**Comment by tldahlgren** [comment]
@vvolkl This is a simple change and, as maintainer, do you want to weigh in before the merge?

**Comment by alecbcs** [comment]
@tldahlgren I'd go ahead and merge this, we already got approval from one of the maintainers (@jmcarcell)

**Review by vvolkl — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**
